api/core/rag/pipeline/queue.py

```diff
@@ -0,0 +1,90 @@
+import json
+from dataclasses import dataclass
+from typing import Any
+
+from extensions.ext_redis import redis_client
+
+TASK_WRAPPER_PREFIX = "__WRAPPER__:"
+
+
+@dataclass
+class TaskWrapper:
+    data: Any
+
+    def serialize(self) -> str:
+        return json.dumps(self.data, ensure_ascii=False)
+
+    @classmethod
+    def deserialize(cls, serialized_data: str) -> "TaskWrapper":
+        data = json.loads(serialized_data)
+        return cls(data)
+
+
+class TenantSelfTaskQueue:
+    """
+    Simple queue for tenant self tasks, used for tenant self task isolation.
+    It uses Redis list to store tasks, and Redis key to store task waiting flag.
+    Support tasks that can be serialized by json.
+    """
+
+    DEFAULT_TASK_TTL = 60 * 60  # 1 hour
+
+    def __init__(self, tenant_id: str, unique_key: str):
+        self.tenant_id = tenant_id
+        self.unique_key = unique_key
+        self.queue = f"tenant_self_{unique_key}_task_queue:{tenant_id}"
+        self.task_key = f"tenant_{unique_key}_task:{tenant_id}"
```

Comment:
I suggest to use a `_` prefix for private attributes.

For example:


```suggestion
        self._tenant_id = tenant_id
        self._unique_key = unique_key
        self._queue = f"tenant_self_{unique_key}_task_queue:{tenant_id}"
        self._task_key = f"tenant_{unique_key}_task:{tenant_id}"
```